switchboard/base.py

```diff
@@ -63,23 +63,23 @@ def __len__(self):  # pragma: nocover
     def __contains__(self, key):  # pragma: nocover
         return self._model.contains(key)
 
-    def __iter__(self):  # pragma: nocover
-        return self._model.all()
+    def __iter__(self):
+        return self.iterkeys()
 
     def __repr__(self):  # pragma: nocover
         return "<%s>" % (self.__class__.__name__)
 
-    def iteritems(self):  # pragma: nocover
+    def iteritems(self):
         def make_item(model):
-            return (getattr(self._key, model), model)
+            return (getattr(model, self._key), model)
```

Comment:
Because I had the wrong param order before. It should be `getattr(object, name[, default])` and I was passing the name (`self._key`) first instead of the object. Once this bit of code was exercised by unit tests, the bug came to light.